### PR TITLE
Safeguards GetAllContents and makes it throw a warning

### PR DESCRIPTION
**Pull request by ninjanomnom**
@MrStonedOne

### CODE REVIEW

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -508,14 +508,16 @@ Recursively gets all contents of contents and returns them all in a list.
 
 recursive_depth is useful if you only want a turf and everything on it (recursive_depth=1)
 */
-/atom/proc/GetAllContents(list/output=list(), recursive_depth=INFINITY)
+/atom/proc/GetAllContents(list/output=list(), recursive_depth=199, _current_depth=0)	// 200 proc calls in and we break shit
```
Comment (optimumtact) [inline code/__HELPERS/unsorted.dm]: this should be a define MAX_PROC_DEPTH

File: code/__HELPERS/unsorted.dm [hunk 2]
```diff
@@ -508,14 +508,16 @@ Recursively gets all contents of contents and returns them all in a list.
...
 	. = output
 	output += src 
-	if(!recursive_depth)
+	if(_current_depth == recursive_depth)
+		if(_current_depth == 199)
```
Comment (optimumtact) [inline code/__HELPERS/unsorted.dm]: use the define

File: code/__HELPERS/unsorted.dm (hunk 1 above)
Comment (MrStonedOne) [reply code/__HELPERS/unsorted.dm]: Also, I'm not sure why you added a new arg when decreasing recusive_depth would be better, faster, and cheaper memory wise.

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Above

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**